src/search/merge_and_shrink/transition_system.cc

```diff
@@ -196,18 +240,24 @@ unique_ptr<TransitionSystem> TransitionSystem::merge(
                     int target2 = transition2.target;
                     int src = src1 * multiplier + src2;
                     int target = target1 * multiplier + target2;
-                    new_transitions.push_back(Transition(src, target));
+                    new_transitions.emplace_back(src, target);
                 }
             }
 
             // Create a new group if the transitions are not empty
-            vector<int> &new_labels = bucket.second;
+            LabelGroup &new_labels = bucket.second;
             if (new_transitions.empty()) {
                 dead_labels.insert(dead_labels.end(), new_labels.begin(), new_labels.end());
             } else {
                 sort(new_transitions.begin(), new_transitions.end());
-                label_groups.push_back(move(new_labels));
-                transitions_by_group_id.push_back(move(new_transitions));
+                sort(new_labels.begin(), new_labels.end());
+                int new_local_label = local_label_infos.size();
+                int cost = INF;
+                for (int label : new_labels) {
+                    cost = min(ts1.labels.get_label_cost(label), cost);
+                    label_to_local_label[label] = new_local_label;
+                }
```

Comment:
This is one of the places where we wouldn't need to compute the cost, but only update the mapping, if the constructor computed cost.